src/PlatformAdapter/Native/Exec/NativeExecImpl.cpp

```diff
@@ -0,0 +1,253 @@
+#include "NativeExecImpl.hpp"
+
+#include "Utils/Utils.hpp"
+
+#undef INFO
+#define INFO(msg, ...)
+
+#undef DEBUG
+#define DEBUG(msg, ...)
+
+/*
+ * NativeExecImpl
+ */
+bool NativeExecImpl::schedule_eval(const std::string& code)
+{
+  if (m_state == DEAD)
+  {
+    FAILED("#NativeExecImpl::schedule_eval, error, dead state");
+    return false;
+  }
+
+  while (m_state != RUNNING)
+  {
+  }
+
+  NativeEvalTask* task = new NativeEvalTask();
+  task->m_code = code;
+  task->m_exec_impl_ptr = this;
+
+  const std::lock_guard<std::mutex> lock(m_tasks_mutex);
+  m_tasks.insert(task);
+
+  uv_async_init(m_loop,
+                &task->m_async_task,
+                [](uv_async_t* async)
+                {
+                  uv_close(reinterpret_cast<uv_handle_t*>(async), nullptr);
+                  NativeEvalTask* task = node::ContainerOf(&NativeEvalTask::m_async_task, async);
```

Comment:
why is this line necessary?

Reply:
compile error when the lambda passed to uv_async_init captures variable.